src/Samples/Sample.AspNetCore/Controllers/PaymentController.cs

```diff
@@ -83,24 +84,44 @@ public async Task<ActionResult> PaymentOrderCancel(string paymentOrderId)
         {
             try
             {
-                var payment = await this.swedbankPayClient.Payments.CardPayments.Get(new Uri(paymentId, UriKind.RelativeOrAbsolute));
 
-                if (payment.Operations.Cancel != null)
-                {
-                    var cancelRequest =
-                        new SwedbankPay.Sdk.Payments.CardPayments.CardPaymentCancelRequest(
-                            DateTime.Now.Ticks.ToString(), "Cancelling parts of the total amount");
-                    var response = await payment.Operations.Cancel(cancelRequest);
-                    TempData["CancelMessage"] = $"Payment has been cancelled: {response.Cancellation.Transaction.Id}";
-                    this.cartService.PaymentOrderLink = null;
-                }
-                else
+                switch (instrument)
                 {
-                    TempData["ErrorMessage"] = "Operation not available";
+                    case PaymentInstrument.CreditCard:
+                        var payment =  await this.swedbankPayClient.Payments.CardPayments.Get(new Uri(paymentId, UriKind.RelativeOrAbsolute));
+
+                        if (payment.Operations.Cancel != null)
+                        {
+                            var cancelRequest = new SwedbankPay.Sdk.Payments.CardPayments.CardPaymentCancelRequest(DateTime.Now.Ticks.ToString(), "Cancelling parts of the total amount");
+                            var response = await payment.Operations.Cancel(cancelRequest);
+                            TempData["CancelMessage"] = $"Payment has been cancelled: {response.Cancellation.Transaction.Id}";
+                            this.cartService.PaymentOrderLink = null;
+                        }
+                        else
+                        {
+                            TempData["ErrorMessage"] = "Operation not available";
+                        }
+                        break;
+                    case PaymentInstrument.Trustly:
+                        var trustlyPayment = await this.swedbankPayClient.Payments.TrustlyPayments.Get(new Uri(paymentId, UriKind.RelativeOrAbsolute));
+
+                        if (trustlyPayment.Operations.Cancel != null)
+                        {
+                            var cancelRequest = new SwedbankPay.Sdk.Payments.TrustlyPayments.TrustlyPaymentCancelRequest(DateTime.Now.Ticks.ToString(), "Cancelling parts of the total amount");
+                            var response = await trustlyPayment.Operations.Cancel(cancelRequest);
+                            TempData["CancelMessage"] = $"Payment has been cancelled: {response.Cancellation.Transaction.Id}";
+                            this.cartService.PaymentOrderLink = null;
+                        }
+                        else
+                        {
+                            TempData["ErrorMessage"] = "Operation not available";
+                        }
```

Comment:
Could this be moved to a helper class/method to reduce the size of the controller? Makes it easier to test for later, and having things seperated makes it easier to follow 😄